registry/storage/blobstore.go

```diff
@@ -90,6 +96,158 @@ func (bs *blobStore) Put(ctx context.Context, mediaType string, p []byte) (distr
 	}, bs.driver.PutContent(ctx, bp, p)
 }
 
+// Enumerate returns an array of digests of all blobs in registry's blob store.
+// Blob's data may or may not be present.
+func (bs *blobStore) Enumerate(ctx context.Context) ([]digest.Digest, error) {
+	return enumerateBlobDigests(ctx, bs.driver, path.Join(storagePathRoot, storagePathVersion, "blobs"), true)
+}
+
+// enumerateBlobDigests is a utility function for walking blob store under given rootPath.
+// If multilevel is true, following layouts are expected:
+//
+//    <rootPath>/<algorithm>/<first two bytes of digest>/<full digest>
+// 	  <rootPath>/tarsum/<version>/<digest algorithm>/<first two bytes of digest>/<full digest>
+//
+// Otherwise following layouts are presumed:
+//
+//    <rootPath>/<algorithm>/<full digest>
+// 	  <rootPath>/tarsum/<version>/<digest algorithm>/<full digest>
+func enumerateBlobDigests(ctx context.Context, d driver.StorageDriver, rootPath string, multilevel bool) ([]digest.Digest, error) {
```

Comment:
I'm not sure I understand the usecase for not using the `multilevel` flag - the multilevel layout is what is laid out by the registry today.  Furthermore I think using the `Walk` function with file patter matching would achieve the same result with less code.